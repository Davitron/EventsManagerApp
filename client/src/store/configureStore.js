import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from '../reducers/rootReducer';

const configureStore = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

configureStore.subscribe(() => {
  console.log('STORE SUB...', configureStore.getState());
});

export default configureStore;
