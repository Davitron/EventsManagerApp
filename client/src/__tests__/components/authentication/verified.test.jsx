import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedVerifiedEmail, { VerifiedEmail } from '../../../components/authentication/VerifiedEmail';

let mountedComponent;

let props;
const locations = [];
const history = {
  push(location) {
    locations.push(location);
  },
};

const dispatch = fn => fn;

const stateProps = {
  isAuthenticated: false,
  status: null,
  data: null
};

const mockStore = configureStore();

let wrapper;


/**
 * @description Initialise the component
 *
 * @returns {object} mountedComponent - Mounte
 */
const getComponent = () => {
  if (!mountedComponent) {
    props = {
      dispatch,
      stateProps
    };
    history.push = jest.fn();
    mountedComponent = shallow(<VerifiedEmail {...props} />);
  }
  return mountedComponent;
};

describe('Verified Component', () => {
  beforeEach(() => {});

  it('component successfully rendered', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      stateProps: {
        status: 'failed',
        data: 'error in registration'
      }
    };

    wrapper = getComponent();

    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().message).toEqual(nextProps.stateProps.data);
  });

  it('testing component will recieve props', () => {
    const nextProps = {
      stateProps: {
        status: 'failed',
        data: 'error in registration'
      }
    };

    wrapper = getComponent();
    wrapper.setProps({
      stateProps: {}
    });
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(wrapper.state().message).toEqual(nextProps.stateProps.data);
  });
});

describe('Connected Component', () => {
  describe('Connected VerifiedEmail component', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        stateProps
      });
      wrapper = shallow(<ConnectedVerifiedEmail store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
