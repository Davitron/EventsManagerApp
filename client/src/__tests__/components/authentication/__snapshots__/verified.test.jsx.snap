// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verified Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerifiedEmail
    dispatch={[Function]}
    stateProps={
      Object {
        "data": null,
        "isAuthenticated": false,
        "status": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="home"
      >
        <div
          className="section section__hero"
          style={
            Object {
              "color": "white",
            }
          }
        >
          <div
            className="my-container"
            style={
              Object {
                "paddingTop": "8em",
              }
            }
          >
            <h2
              className="animated fadeInUp"
            >
              
            </h2>
            <h3
              className="animated fadeInUp"
              style={
                Object {
                  "paddingBottom": "60px",
                }
              }
            >
              <Link
                replace={false}
                to="/login"
              >
                Proceed to login
              </Link>
            </h3>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="section section__hero"
          style={
            Object {
              "color": "white",
            }
          }
        >
          <div
            className="my-container"
            style={
              Object {
                "paddingTop": "8em",
              }
            }
          >
            <h2
              className="animated fadeInUp"
            >
              
            </h2>
            <h3
              className="animated fadeInUp"
              style={
                Object {
                  "paddingBottom": "60px",
                }
              }
            >
              <Link
                replace={false}
                to="/login"
              >
                Proceed to login
              </Link>
            </h3>
          </div>
        </div>,
        "className": "home",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="my-container"
            style={
              Object {
                "paddingTop": "8em",
              }
            }
          >
            <h2
              className="animated fadeInUp"
            >
              
            </h2>
            <h3
              className="animated fadeInUp"
              style={
                Object {
                  "paddingBottom": "60px",
                }
              }
            >
              <Link
                replace={false}
                to="/login"
              >
                Proceed to login
              </Link>
            </h3>
          </div>,
          "className": "section section__hero",
          "style": Object {
            "color": "white",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="animated fadeInUp"
              >
                
              </h2>,
              <h3
                className="animated fadeInUp"
                style={
                  Object {
                    "paddingBottom": "60px",
                  }
                }
              >
                <Link
                  replace={false}
                  to="/login"
                >
                  Proceed to login
                </Link>
              </h3>,
            ],
            "className": "my-container",
            "style": Object {
              "paddingTop": "8em",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "animated fadeInUp",
              },
              "ref": null,
              "rendered": "",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/login"
                >
                  Proceed to login
                </Link>,
                "className": "animated fadeInUp",
                "style": Object {
                  "paddingBottom": "60px",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Proceed to login",
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Proceed to login",
                "type": [Function],
              },
              "type": "h3",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="home"
        >
          <div
            className="section section__hero"
            style={
              Object {
                "color": "white",
              }
            }
          >
            <div
              className="my-container"
              style={
                Object {
                  "paddingTop": "8em",
                }
              }
            >
              <h2
                className="animated fadeInUp"
              >
                
              </h2>
              <h3
                className="animated fadeInUp"
                style={
                  Object {
                    "paddingBottom": "60px",
                  }
                }
              >
                <Link
                  replace={false}
                  to="/login"
                >
                  Proceed to login
                </Link>
              </h3>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="section section__hero"
            style={
              Object {
                "color": "white",
              }
            }
          >
            <div
              className="my-container"
              style={
                Object {
                  "paddingTop": "8em",
                }
              }
            >
              <h2
                className="animated fadeInUp"
              >
                
              </h2>
              <h3
                className="animated fadeInUp"
                style={
                  Object {
                    "paddingBottom": "60px",
                  }
                }
              >
                <Link
                  replace={false}
                  to="/login"
                >
                  Proceed to login
                </Link>
              </h3>
            </div>
          </div>,
          "className": "home",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="my-container"
              style={
                Object {
                  "paddingTop": "8em",
                }
              }
            >
              <h2
                className="animated fadeInUp"
              >
                
              </h2>
              <h3
                className="animated fadeInUp"
                style={
                  Object {
                    "paddingBottom": "60px",
                  }
                }
              >
                <Link
                  replace={false}
                  to="/login"
                >
                  Proceed to login
                </Link>
              </h3>
            </div>,
            "className": "section section__hero",
            "style": Object {
              "color": "white",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="animated fadeInUp"
                >
                  
                </h2>,
                <h3
                  className="animated fadeInUp"
                  style={
                    Object {
                      "paddingBottom": "60px",
                    }
                  }
                >
                  <Link
                    replace={false}
                    to="/login"
                  >
                    Proceed to login
                  </Link>
                </h3>,
              ],
              "className": "my-container",
              "style": Object {
                "paddingTop": "8em",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "className": "animated fadeInUp",
                },
                "ref": null,
                "rendered": "",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    replace={false}
                    to="/login"
                  >
                    Proceed to login
                  </Link>,
                  "className": "animated fadeInUp",
                  "style": Object {
                    "paddingBottom": "60px",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Proceed to login",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Proceed to login",
                  "type": [Function],
                },
                "type": "h3",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
